tests/integ/test_simplifier_suite.py

```diff
@@ -331,6 +333,130 @@ def test_select_expr(session, simplifier_table):
     assert df11.queries["queries"][-1].count("SELECT") == 2
 
 
+def test_select_with_table_function_join(session):
+    # setup
+    df = session.create_dataframe(
+        [[1, 2, "one o two"], [2, 3, "two o three"]], schema=["a", "b", "c"]
+    )
+    split_to_table = table_function("split_to_table")
+
+    @udtf(output_schema=["two_x", "three_x"])
+    class multiplier_udtf:
+        def process(self, n: int) -> Iterable[Tuple[int, int]]:
+            yield (2 * n, 3 * n)
+
+    df1 = df.select("a", split_to_table("c", lit(" ")))
+    df2 = df.select("a", multiplier_udtf(df.b))
+    # test multiple selects are flattened
+    df3 = df1.select("a", "seq").select("a")
+    assert df1.queries["queries"][-1].count("SELECT") == df3.queries["queries"][
```

Comment:
You should assert that the result equals a value?

Reply:
I am adding output equals checks for repeated select. rest are tested in regular tests

Reply:
I see